tests/unit/package_managers/yarn/test_resolver.py

```diff
@@ -436,7 +481,11 @@ def mock_project(project_dir: RootedPath) -> Project:
                 packjson_path="node_modules/strip-ansi-tarball/package.json",
                 packjson_content=json.dumps({"name": "strip-ansi"}),
             ),
-            Component(name="strip-ansi", version="4.0.0", purl=""),
+            Component(
+                name="strip-ansi",
+                version="4.0.0",
+                purl=f"pkg:npm/strip-ansi@4.0.0?vcs_url={MOCK_REPO_VCS_URL}#external-packages/strip-ansi-4.0.0.tgz",
```

Comment:
Is it worth adding a scenario that covers packages that live in parent folders? Maybe we missed it the first time because there was no scenario showing it.

Reply:
just added the scenario, PTAL